semantic-java/src/Language/Java.hs

```diff
@@ -0,0 +1,17 @@
+-- | Semantic functionality for Java programs.
+module Language.Java
+( Term(..)
+) where
+
+import qualified Language.Java.Tags as JavaTags
+import qualified Tags.Tagging.Precise as Tags
+import qualified TreeSitter.Java.AST as Java
+import qualified TreeSitter.Unmarshal as TS
+
+newtype Term a = Term { getTerm :: Java.Program a }
```

Comment:
As with python, we define a `Term` type as a convenient, consistent moniker so that:

1. no other code needs to know whether it’s `Module` or `Program` or something else for any particular language;
2. we have a convenient type to define instances for (it’s likely inconvenient to define them for the precise AST types since they’re defined in the `tree-sitter-*` packages); and
3. especially for small, homoiconic languages, we have a clear delineation between the top-level term representing the contents of a file, and terms which might be able to occur at many positions within a file.